packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -4572,10 +4572,12 @@ function invokeLayoutEffectMountInDEV(fiber: Fiber): void {
       }
       case ClassComponent: {
         const instance = fiber.stateNode;
-        try {
-          instance.componentDidMount();
-        } catch (error) {
-          captureCommitPhaseError(fiber, fiber.return, error);
+        if (typeof instance.componentDidMount === 'function') {
```

Comment:
why is this needed?

Reply:
The PR always sets `MountLayoutDev` on mount regardless if `instance.componentDidMount` exists, and when the flag is set, this code path always runs.